src/reactjs/tests/utils.tsx

```diff
@@ -24,10 +33,10 @@ export function mockNavigatorOnLine(value: boolean) {
   return jest.spyOn(navigator, 'onLine', 'get').mockReturnValue(value)
 }
 
-export function mockConsoleError() {
-  const consoleMock = jest.spyOn(console, 'error')
-  consoleMock.mockImplementation(() => undefined)
-  return consoleMock
+export const mockLogger = {
+  log: jest.fn(),
+  warn: jest.fn(),
+  error: jest.fn(),
```

Comment:
why do we need he mockLogger if the defaultLogger defaults to `jest.fn()` in test env?

Reply:
Default logger (i.e. the return value of `getDefaultLogger`) doesn't default to mock logger/jest.fn because it's tricky to dead code eliminite `jest.fn` calls in non-test environments. So the reason default logger doesn't use jest.fn() in test environments is because we don't want to refer to the global `jest` variable in files that go into production development/builds.

Reply:
but, it does, for `process.env.NODE_ENV === 'test'`, or am I missing something?

https://github.com/tannerlinsley/react-query/pull/3246/files#diff-b26e2d2c63ca623f55f84e869194f983f9dbb81fd0703cf38e5f11edc091f5c3R11-R18

Reply:
I tried that approach and gave up for the reasons I give above. It seems I forgot to remove that code, my bad 😸